aea/cli/reset_password.py

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2020 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""Implementation of the 'aea reset_password' subcommand."""
+
+import click
+
+from aea.cli.registry.login import registry_reset_password
+
+
+@click.command(name="reset_password", help="Reset password of Registry account.")
+@click.argument("email", type=str, required=True)
+def reset_password(email):
+    """Command to request Registry to reset password."""
+    _do_password_reset(email)
+
+
+def _do_password_reset(email: str) -> None:
+    """
+    Request Registry to reset password.
+
+    :param email: str email.
+
+    :return:
+    """
+    registry_reset_password(email)
```

Comment:
The following command:
```
aea reset_password this-is-a-wrong-email   
```

Gave me:
```
Error: {'email': ['Enter a valid email address.']}
```

That means the request was sent, with a wrong email.
In my opinion, the validation of the content of `email` should be done locally, and not delegated to the remote APIs. 

- First of all, because the command would be more responsive. 
- Also, because we give the right feedback to the user; the command failed because the input was wrong, not for connectivity issues. 
- Although not a problem at the current state of the situation, we reduce some avoidable workload to the server.
- I also think it's a better approach on the security side of things. We allow requests with arbitrary content as arguments, which may break both the client and the server unexpectedly (of course we won't solve all the security risks on the server-side, but at least the official clients behave correctly). This is part of a broader set of issues, some of them included into #1727.

Anyway, I don't think it's a priority, but I suggest to keep track of it in that case.

Reply:
@marcofavorito can I ask you for assistance with that? You've implemented a PublicIdParameter that validates Public IDs, can you do the same for email? In separate ticket/pr

Reply:
Yes, sure